apache/superset
# fix: Drill to detail blocked by tooltip

### SUMMARY
Implements a custom tooltip positioning function that ensures that the tooltip never overlaps with the pointer during normal use. In addition this fixes a bug that would cause the tooltip to stay on screen if the tooltip was visible when the user navigated away from the page.

This PR also cleans up some types on the ECharts plugin package (these kinda needed to be fixed as the code was becoming difficult to manage.

### AFTER
Now tooltips are allowed to overflow other charts:

https://user-images.githubusercontent.com/33317356/203587143-d3604907-5412-4d9f-96c1-415bdcff2439.mp4

In addition, the sticky hover bug is also gone:

https://user-images.githubusercontent.com/33317356/203587757-8f91b009-b68d-403c-b425-de9cdecb5738.mp4

### BEFORE
Before, the pointer would overflow the tooltip if the chart was small enough and never utilized free space next to the chart:

https://user-images.githubusercontent.com/33317356/203589170-ceb21fe3-4745-443e-bf7f-2ee8209dd407.mp4

Also navigating away from the page with the chart would sometimes leave the tooltip stuck in the middle of the screen:

https://user-images.githubusercontent.com/33317356/203589238-961925bb-9f31-4326-8f3d-aa3b3d013dcb.mp4



### TESTING INSTRUCTIONS
1. Go to eph env and open the "tooltip test" dashboard.
2. Try to make the tooltip overlap the pointer of fall outside the screen (note that scrolling can still make the tooltip go outside)
3. Navigate away from the dashboard page while the tooltip is open (by using keyboard shortcuts) - the tooltip will no longer be stuck on the screen after navigation.

### ADDITIONAL INFORMATION


:
